File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode node = new ListNode();
        node.next = head;
        ListNode fast = head;
        
        int len = 0;
        while(fast!=null){
            len+=1;
            fast=fast.next;
        }
        
        if(len-n==0){
            return node.next.next;
        }
        
        System.out.println(len);
        
        int currentPos = 0;
        while(head!=null){
            if(currentPos+1==(len-n)){
                head.next = head.next.next;
            }
            currentPos+=1;
            head = head.next;
        }
        return node.next;
        
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        length = 0
        countNode = head
        while countNode:
            countNode = countNode.next
            length+=1
        
        currPos = 0
        dummy = ListNode()
        dummy.next = head
        resultNode = dummy
        while dummy:
            if currPos==length-n:
                dummy.next = dummy.next.next
                break
            dummy = dummy.next
            currPos+=1
        return resultNode.next
                
        